src/components/products/ProductCard.tsx

```diff
@@ -0,0 +1,63 @@
+import { HeartIcon } from "../icons";
+import { Product } from "../../types";
+import { useState, useEffect } from "react";
+
+export const ProductCard = ({
+  title,
+  description,
+  price,
+  images,
+  id,
+}: Product) => {
+  const [isFav, setIsFav] = useState(false);
+
+  useEffect(() => {
+    const favorites = localStorage.getItem("favorites");
+    const favoritesArray = favorites ? JSON.parse(favorites) : [];
+
+    const productExists = favoritesArray.find(
+      (item: Product) => item.id === id
+    );
+    setIsFav(!!productExists);
+  }, [id]);
+
+  const handleFav = () => {
+    const product = { title, description, price, images, id };
+    const favorites = localStorage.getItem("favorites");
```

Comment:
you can extract this as it is used here and on line 15 and create a helper function:
```
const getFavorites = () => { 
     const favorites = localStorage.getItem("favorites");
      return favorites ? JSON.parse(favorites) : [];
}
```